apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -318,20 +340,31 @@ impl ApplyTo for PathSelection {
                             json_type_name(data),
                         )
                         .as_str(),
-                        input_path,
+                        input_path_with_key.to_vec(),
                     ));
                     None
-                };
-
-                input_path.pop();
-
-                result
+                }
             }
-            Self::Selection(selection) => {
-                // If data is not an object here, this recursive apply_to_path
-                // call will handle the error.
-                selection.apply_to_path(data, vars, input_path, errors)
+            Self::Method(method_name, method_args, tail) => {
+                if let Some(method) = ARROW_METHODS.get(method_name) {
+                    method(
+                        method_name.as_str(),
+                        method_args,
+                        data,
+                        vars,
+                        input_path,
+                        tail.as_ref(),
+                        errors,
+                    )
+                } else {
+                    errors.insert(ApplyToError::new(
+                        format!("Method ->{} not found", method_name).as_str(),
+                        input_path.to_vec(),
+                    ));
```

Comment:
We definitely don't want missing methods to end up as a runtime error. This combined with the fact that validation and expansion will _need_ to handle each of the built-in methods differently, depending on their signature, makes me think `ARROW_METHODS` should actually be a `Methods` enum for now that we can match against to make sure we're handling them all properly. If/when we _need_ dynamic methods, we could figure that out.

Reply:
We could check `ARROW_METHODS` during parsing to make sure any `->` method name encountered by the parser is known. That should make the method vocabulary statically enforceable, instead of only dynamically enforceable.